src/Apps/Order2/Routes/Checkout/Hooks/useReturnToArtwork.ts

```diff
@@ -0,0 +1,41 @@
+import { Order2CheckoutContext } from "Apps/Order2/Routes/Checkout/CheckoutContext/Order2CheckoutContext"
+import {
+  handleBackNavigation,
+  preventHardReload,
+} from "Apps/Order2/Utils/navigationGuards"
+import { useCallback } from "react"
+
+interface ReturnToArtworkOptions {
+  /* Unload navigation blockers before navigation */
+  force?: boolean
+}
+
+export function useReturnToArtwork() {
+  // Get router and orderData directly from the store to avoid action wrapper types
+  const router = Order2CheckoutContext.useStoreState(state => state.router)
+  const orderData = Order2CheckoutContext.useStoreState(
+    state => state.orderData,
+  )
+  const artworkSlug = orderData?.lineItems?.[0]?.artwork?.slug
+
+  const returnToArtwork = useCallback(
+    ({ force }: ReturnToArtworkOptions) => {
+      if (!artworkSlug) {
+        console.warn("No artwork slug found, cannot return to artwork")
+        return
+      }
+
+      // For web, navigate directly to the artwork
+      if (force) {
+        // Remove navigation guards before navigating
+        window.removeEventListener("beforeunload", preventHardReload)
+        window.removeEventListener("popstate", handleBackNavigation)
+      }
+
+      router.push(`/artwork/${artworkSlug}`)
+    },
+    [artworkSlug, router],
+  )
```

Comment:
To clean up. initially (first commit as of now) i tried to include eigen-specific logic for [EMI-3010] but decided that was overkill and would introduce new issues.